exporters/autoexport/metrics.go

```diff
@@ -94,4 +100,53 @@ func init() {
 	RegisterMetricReader("none", func(ctx context.Context) (metric.Reader, error) {
 		return newNoopMetricReader(), nil
 	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		// variable names and defaults specified at https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#prometheus-exporter
+		host := getenv("OTEL_EXPORTER_PROMETHEUS_HOST", "localhost")
+		port := getenv("OTEL_EXPORTER_PROMETHEUS_PORT", "9464")
+
+		mux := http.NewServeMux()
+		reg := prometheus.NewRegistry()
+		mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
+
+		// Timeouts are necessary to make a server resilent to attacks, but
+		// ListenAndServe doesn't set any. We use timeout values from this example: https://blog.cloudflare.com/exposing-go-on-the-internet/#:~:text=There%20are%20three%20main%20timeouts
+		// TODO: find defaults with a better justification, and/or make these configurable
+		go serve(ctx, &http.Server{
+			Addr:         host + ":" + port,
```

Comment:
Can we use [`net.Listen`](https://pkg.go.dev/net#Listen) before starting the goroutine so that we would return earlier if the provided address or port is invalid?

The we will  use `server.Listen` instead of `server.ListenAndServe` and we would not need to set `server.Addr`.

Reply:
Done.